spec/models/customer_spec.rb

```diff
@@ -68,6 +68,112 @@
     end
   end
 
+  describe '#display_name' do
+    subject(:customer) { build(:customer, name:, legal_name:, firstname:, lastname:) }
+
+    let(:name) { 'ACME Inc' }
+    let(:legal_name) { 'ACME International Corporation' }
+    let(:firstname) { 'Thomas' }
+    let(:lastname) { 'Anderson' }
+
+    context 'when all fields are nil' do
+      let(:name) { nil }
+      let(:legal_name) { nil }
+      let(:firstname) { nil }
+      let(:lastname) { nil }
+
+      it 'returns an empty string' do
+        expect(customer.display_name).to eq('')
+      end
+    end
+
+    context 'when name and legal_name are nil' do
+      let(:name) { nil }
+      let(:legal_name) { nil }
+
+      it 'returns only firstname and lastname if present' do
+        expect(customer.display_name).to eq('Thomas Anderson')
+      end
+    end
+
+    context 'when firstname and lastname are nil' do
+      let(:firstname) { nil }
+      let(:lastname) { nil }
+
+      it 'returns only the name when prefer_legal_name is false' do
+        expect(customer.display_name(prefer_legal_name: false)).to eq('ACME Inc')
+      end
+
+      it 'returns only the legal_name when prefer_legal_name is true' do
+        expect(customer.display_name(prefer_legal_name: true)).to eq('ACME International Corporation')
+      end
+    end
+
+    context 'when name is present and both firstname and lastname are present' do
+      let(:legal_name) { nil }
+
+      it 'returns only firstname and lastname when prefer_legal_name is true' do
+        expect(customer.display_name(prefer_legal_name: true)).to eq('Thomas Anderson')
+      end
+
+      it 'returns name with firstname and lastname when prefer_legal_name is false' do
+        expect(customer.display_name).to eq('ACME Inc - Thomas Anderson')
+      end
+    end
+
+    context 'when legal_name is present and both firstname and lastname are present' do
+      let(:name) { nil }
+
+      it 'returns legal_name with firstname and lastname when prefer_legal_name is true' do
+        expect(customer.display_name(prefer_legal_name: true)).to eq('ACME International Corporation - Thomas Anderson')
```

Comment:
Same here: We can change these spec now that we're not using `prefer_legal_name`.